### ISSUE TITLE
[req]: Azusa

### ISSUE THREAD

**Issue by RocksonLee** [Jackett/Jackett#14478]
### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Is there already a request for your tracker?

- [X] I have checked older issues, open and closed

### Type

Private

### Tracker details

```markdown
- Name: Azusa
- URL: https://azusa.wiki
- Description:  Azusa is a CHINESE Torrent Tracker focusing on comic
- Minimum Seeding Time and Minimum Ratio (if Private): 
- Additional Comments:
```
Registration is currently closed, invite only.
Engine: NexusPHP v1.7.20 2022-08-05

**Comment by garfield69** [comment]
As registration is currently invite only, then:
As long as the site does not have an entrance exam where you have to upload and download to pass, then if you send an invite to garfieldsixtynine -at- gmail.com I will build the cardigann indexer.
If there is an exam then please send the invite to jacketttest -at- gmail.com

Alternatively, given that the site is using nexusphp we can usually build an indexer using a template, but we will need to see the html content of the torrents.php page in order to set the categories and crosscheck the field selectors.
See https://github.com/Jackett/Jackett/wiki/Troubleshooting#how-to-provide-html-source

**Comment by garfield69** [comment]
Nice. I'll work on this in my next free-time slot ;-)

**Comment by garfield69** [comment]
Can you check if the site has any rules or Hit & Run information about how long you should seed a torrent after a download, and/or if you have to reach a ratio, for example 24 hours seed time, and/or a ratio of 1 to 1

**Comment by garfield69** [comment]
v0.21.265

**Comment by garfield69** [comment]
@RocksonLee  https://azusa.wiki/ redirects me to https://www.baidu.com/
Is the indexer working for you?
is the site GONE, just BROKEN, or moved to a new DOMAIN?

**Comment by ilike2burnthing** [comment]
They might have just disabled the login page, some links 'work' (they don't redirect to Baidu):
- https://azusa.wiki/favicon.ico - working favicon
- https://azusa.wiki/about.php - file not found
- https://azusa.wiki/login - 404 | NOT FOUND

There are also 2 login pages for image hosting accounts:
- https://img.azusa.wiki/login
- https://i.azusa.wiki/login

And their mail page:
- https://mail.azusa.wiki/ - ERR_CONNECTION_REFUSED

The other option is that the tracker is dead, but the owner has kept the TLD for their personal use. Your guess is as good as mine.

**Comment by luckylemonade24** [comment]
> 他们可能刚刚禁用了登录页面，一些链接 “有效” （他们不会重定向到百度）：
> 
> * https://azusa.wiki/favicon.ico - 工作网站图标
> * https://azusa.wiki/about.php - 找不到文件
> * https://azusa.wiki/login - 404 |未找到
> 
> 镜像托管账户还有 2 个登录页面：
> 
> * https://img.azusa.wiki/login
> * https://i.azusa.wiki/login
> 
> 以及他们的邮件页面：
> 
> * https://mail.azusa.wiki/ - ERR_CONNECTION_REFUSED
> 
> 另一种选择是跟踪器已死，但所有者保留了 TLD 供其个人使用。我也不知道。

我在各pt站内论坛寻找邀请，请问现在还开放邀请吗，22年就知道这个站了，一直没找到进入方法

**Comment by garfield69** [comment]
@luckylemonade24  This GitHub repository is a support site for the Jackett Indexer and not a forum to solicit invitations to a web site. Please look elsewhere.

**Comment by garfield69** [comment]
Yes, the presence of https://azusa.wiki/favicon.ico indicates that the site is still there, so it must be as you say, login page disabled.
What surprises me is that no one has complained that the indexer is not working %-/
So if its not broken then don't touch it.

**Comment by christaikobo** [comment]
@garfield69 
I'm on the site so here are some contexts:

They have been hit by DNS poisoning a lot just like most of other Chinese trackers.  I guess because of it they are quite paranoid and often shutdown the login page during politically sensitive times like when the CCP hold party meetings or June 4th or whatever. When they close down, you can either use cookie to keep login status, or bind your account to their official telegram bot to get a one time login link with token. When they shutdown the login page redirects to Baidu.com.

The reason why nobody complains is probably because they are very small and niche, barely anyone from the English community is on the site. 

I'm using prowlarr and they seem to be using the jackett indexer so I'm sorta of incentivized to get it working if possible. It is not working right now. 

I have always wondered why you guys don't use cookie for this site, but rather opt to go with login. 

But anyway, shoot away if you have questions and need help. But unfortunately I don't have the privilege to send invites. I can help you guys test though. I can spin up a test docker of jackett if need be.

**Comment by garfield69** [comment]
I will switch to the cookie method and it will get published in the next release of Jackett sue out in about 10 hours or so.
if you could test that would be great.

**Comment by christaikobo** [comment]
Sure no problem. Thanks!

**Comment by garfield69** [comment]
v0.22.1336

**Comment by christaikobo** [comment]
I got `Login failed: Selector "a[href="logout.php"]" didn't match` when adding the indexer. 

They do say that page layout was optimized on 24th of Jan. Also CSRF verification was added on 16th of Jan.

How should I proceed?

**Comment by garfield69** [comment]
Thanks for the latest torrents page.
it does show that the logout link we are expecting is different.
I'll update the indexer and in another 13 hours or so please try again and see if its solved.

**Comment by christaikobo** [comment]
Of course, thanks!

**Comment by garfield69** [comment]
v0.22.1345

**Comment by christaikobo** [comment]
Indexer added successfully, but search doesn't work. I think I know the reason, please check your Email.

**Comment by garfield69** [comment]
So without the csrf_token, the indexer returns zero results, or does it generate an error?

**Comment by christaikobo** [comment]
Returns zero result.

**Comment by garfield69** [comment]
Please turn on Jackett enhanced logging and repeat the test and provide the log. Hopefully the debug and HTML will provide a clearer picture.
Cardigann does not have fetch-token-before-search logic, so if it is really needed for the torrent search url then we would have to rewrite the indexer in C#, which ends my contribution to this indexer since i am not a C# dev :(

**Comment by christaikobo** [comment]
I will get you the logs.

Meanwhile, how about a manual input form? The token should be tied to the session and won't change. Just have user manually input the token, will it work?

**Comment by christaikobo** [comment]
Logs sent.

**Comment by garfield69** [comment]
Good news, the problem appears to be that the indexer is not parsing the leechers from the results, so I will fix that and we can try again after the next Jackett release ;-)

**Comment by garfield69** [comment]
v0.22.1355

**Comment by christaikobo** [comment]
Glad to report that it is working!

**Comment by garfield69** [comment]
\o/

**Comment by christaikobo** [comment]
The site seems to have gone through another overhaul. Search yet again returns zero result. I have sent the latest information via Email.

**Comment by garfield69** [comment]
From the log you provided it appears that in response to Jackett sending the search query to the site, it responded with
```
Forbidden
你可能正在通过脚本访问站点，请通过页面进行搜索，如果正常访问遇此提示，请联系管理组汇报。
```
which is what I get If I try using the search path without credentials.

As an experiment, please access the torrents page and perform a search for a title, and take a note of the address the browser has used once results are returned.
If the search path includes `&csrf_token=<token>`
then remove that from the address and see if that generates the error.
Make sure the browser does not autocomplete your address edit by providing the previous csrf parm before pressing enter ;-)
Let me know how it goes.

**Comment by christaikobo** [comment]
Information sent via Email.

**Comment by garfield69** [comment]
What if you also add `&notnewword=1` to the end of the search?
Also, as a sanity check, have you refreshed the cookie on your indexer config before doing a search?

**Comment by christaikobo** [comment]
Information sent via Email.

**Comment by garfield69** [comment]
I've updated the indexer to drop `notnewword` parm, and added some of the other parms that the indexer used to leave out, just in case they are now mandatory.
The next release of Jackett will include the update, due to be published in about 9 hours or so.
Let me know how you go when you next try the indexer out ;-)

**Comment by christaikobo** [comment]
I just manually used your updated `azusa.yml` in prowlarr's custom definition, added the custom indexer, tried a search, it works! 

Thank you!

**Comment by garfield69** [comment]
\o/

**Comment by garfield69** [comment]
v0.22.2219
